src/taskgraph/util/vcs.py

```diff
@@ -157,6 +157,16 @@ def find_latest_common_revision(self, base_ref):
             "{node}",
         ).strip()
 
+    def does_revision_exist_locally(self, revision):
+        try:
+            return self.run("log", "-r", f"ancestors(.) and '{revision}'").strip() != ""
```

Comment:
So looks like on `hg` this only returns True if it exists AND is an ancestor of the current revision. But on Git that's not the case? Why the discrepency? Let's make sure to document the differences here.

Reply:
Oh thank you very much for spotting this 👍 I just fixed the discrepancy by making this function compliant to the git one.